### PR TITLE
Fixes part of #1095: Addition of config file 

### LINKED ISSUE #1095: Addition of Firebase SDK [GSoC'20][Milestone 1]
Setting up Firebase in the application by - 
- Registering the app on Firebase console.
- Adding the Firebase Configuration file `google-services.json` to our codebase.
- Adding basic Firebase dependencies in both project-level and app-level `build.gradle`. 
- Adding `Crashlytics` dependency and setting it up from console as well
- Enabling `Google Analytics` for our application in the Firebase console and adding its dependency in the codebase.

Important Links - 
- [Add Firebase to your Android project](https://firebase.google.com/docs/android/setup)
- [Setting up Crashlytics](https://firebase.google.com/docs/crashlytics/get-started?platform=android)
- [Setting up Analytics](https://firebase.google.com/docs/crashlytics/get-started?platform=android)

### PR DESCRIPTION
**Pull request by Sarthak2601**
## Explanation

Fixes part of #1095 
- Addition of the Firebase's `google-services.json` config file

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by vinitamurthi — APPROVED** [review]
Wait on @BenHenning to approve

File: app/google-services.json
Comment (BenHenning) [inline app/google-services.json]: How many of these values need to be real? Can we check in fake values for now, then fill in the correct values once we have a shared developer-only Firebase project to reference? I plan to create both a developer and production project to make this simpler.

**Review by BenHenning — CHANGES_REQUESTED** [review]

File: app/google-services.json
Comment (Sarthak2601) [reply app/google-services.json]: From what I have seen, the package name should be the same and others can be changed. The only downside I see is that it will throw a lot of errors in the logcat when we use the dummy values, will that be okay ?

**Comment by aggarwalpulkit596** [comment]
Can't we add a application suffix id for debug build ?

**Comment by Sarthak2601** [comment]
> Can't we add a application suffix id for debug build ?
> […](#)

We can and we might as well but that will be done when we set up our own development project in some time.

File: app/google-services.json
Comment (vinitamurthi) [inline app/google-services.json]: I think we should make it obvious that this is fake data..something like "1:1234...:android:fakestring" or something
From this it doesn't look very obvious

**Review by vinitamurthi — CHANGES_REQUESTED** [review]

File: app/google-services.json
Comment (BenHenning) [inline app/google-services.json]: Ditto here--can we clearly indicate that this is a fake string?

File: app/google-services.json
Comment (BenHenning) [inline app/google-services.json]: And for this key.

**Review by BenHenning — CHANGES_REQUESTED** [review]

File: app/google-services.json
Comment (BenHenning) [reply app/google-services.json]: Are these errors just at startup, or repeated? It's not ideal, and we should move to a debug project soon but some short-term errors just on app startup would be fine.

Can you file issues for creating the developer & production projects, and assign them to me? I defer to @vinitamurthi when these should be done by, but I'll make sure my part is done within whatever milestone we bucket them into.

Comment (Sarthak2601) [reply app/google-services.json]: They are repeated errors, not just at the startup. Okay, I'll create those issues.

File: app/google-services.json
Comment (Sarthak2601) [reply app/google-services.json]: Done

File: app/google-services.json
Comment (Sarthak2601) [reply app/google-services.json]: Done

File: app/google-services.json
Comment (Sarthak2601) [reply app/google-services.json]: Done

**Review by vinitamurthi — APPROVED** [review]

**Comment by vinitamurthi** [comment]
Looks good! You probably want to pull from develop once to get the new CircleCI changes

File: app/google-services.json
Comment (vinitamurthi) [reply app/google-services.json]: @BenHenning I think we should allow this PR to go in with the fake data because all of the other PRs that @Sarthak2601 is going to create for this milestone is dependent on this PR (the build fails without this json so there has to be a json in for any of the firebase work to begin). So either we get the development project with the proper data ready ASAP, or we let this PR go in and then you can get the development project ready by maybe 1 week before milestone 1 ends.

Comment (BenHenning) [reply app/google-services.json]: @vinitamurthi will it be fine to get this checked in early next week? I can get the development project created next Monday to unblock this PR. The alternative seems to be pretty bad: repeated errors in logcat.

**Comment by BenHenning** [comment]
@Sarthak2601 the developer project is now available. Please update the JSON file to match that & send back to me once that's done.

**Comment by Sarthak2601** [comment]
> @Sarthak2601 the developer project is now available. Please update the JSON file to match that & send back to me once that's done.

JSON file updated, PTAL.

File: app/google-services.json
Comment (vinitamurthi) [reply app/google-services.json]: Resolving this since we the new project has been created

**Review by BenHenning — APPROVED** [review]
Thanks @Sarthak2601!

**Pull request merged**
